libs/designsystem/src/lib/components/chart/chart-js/chart-js.service.ts

```diff
@@ -30,15 +48,101 @@ export class ChartJSService {
       dataLabels,
       customOptions,
       annotations,
+      datalabelOptions,
       highlightedElements,
     } = args;
 
-    const datasets = this.createDatasets(data, highlightedElements);
+    const suffix = datalabelOptions?.valueSuffix ? datalabelOptions.valueSuffix : '';
+
+    // We need to modify the datasets in order to add tooltips.
+    const datasets =
+      datalabelOptions?.showCurrent || datalabelOptions?.showMax || datalabelOptions?.showMin
+        ? this.createDatasets(this.addTooltips(data, datalabelOptions, suffix), highlightedElements)
+        : this.createDatasets(data, highlightedElements);
+
     const options = this.createOptionsObject({ type, customOptions, annotations });
-    const config = this.createConfigurationObject(type, datasets, options, dataLabels);
+    let config = this.createConfigurationObject(type, datasets, options, dataLabels);
+
+    if (type === ChartTypes.stock) {
+      config.options.plugins.tooltip.callbacks.label = (context) => {
+        return context.formattedValue + suffix;
+      };
+
+      config.options.scales.y.ticks.callback = (value) => {
+        return value + suffix;
+      };
+
+      // assuming that the first dataset controls the datespan.
+      const chartPeriod = this.chartConfigService.findChartPeriod(datasets[0]);
+      config.options = this.handleLocalization(
+        config.options,
+        chartPeriod,
+        datalabelOptions.locale
+      );
+    }
```

Comment:
As we're modifying the options object, I think it would make sense to do this as part of the `createOptionsObject` method as this has the responsibility of setting up the options object.